src/compiler/scala/tools/nsc/transform/patmat/MatchAnalysis.scala

```diff
@@ -81,94 +81,71 @@ trait TreeAndTypeAnalysis extends Debugging {
   trait CheckableTreeAndTypeAnalysis {
     val typer: Typer
 
-    // TODO: domain of other feasibly enumerable built-in types (char?)
-    def enumerateSubtypes(tp: Type, grouped: Boolean): List[List[Type]] =
-      tp.typeSymbol match {
-        // TODO case _ if tp.isTupleType => // recurse into component types?
-        case UnitClass if !grouped =>
-          List(List(UnitTpe))
-        case BooleanClass if !grouped =>
-          List(ConstantTrue :: ConstantFalse :: Nil)
-        // TODO case _ if tp.isTupleType => // recurse into component types
-        case modSym: ModuleClassSymbol if !grouped =>
-          List(List(tp))
-        case sym: RefinementClassSymbol =>
-          val parentSubtypes = tp.parents.flatMap(parent => enumerateSubtypes(parent, grouped))
-          if (parentSubtypes exists (_.nonEmpty)) {
-            // If any of the parents is enumerable, then the refinement type is enumerable.
-            // We must only include subtypes of the parents that conform to `tp`.
-            // See neg/virtpatmat_exhaust_compound.scala for an example.
-            parentSubtypes map (_.filter(_ <:< tp))
-          }
-          else Nil
-        // make sure it's not a primitive, else (5: Byte) match { case 5 => ... } sees no Byte
-        case sym if sym.isSealed =>
-
-          val tpApprox = analyzer.approximateAbstracts(tp)
-          val pre = tpApprox.prefix
-
-          def filterChildren(children: List[Symbol]): List[Type] = {
-            children flatMap { sym =>
-              // have to filter out children which cannot match: see ticket #3683 for an example
-              // compare to the fully known type `tp` (modulo abstract types),
-              // so that we can rule out stuff like: sealed trait X[T]; class XInt extends X[Int] --> XInt not valid when enumerating X[String]
-              // however, must approximate abstract types in
-
-              val memberType = nestedMemberType(sym, pre, tpApprox.typeSymbol.owner)
-              val subTp = appliedType(memberType, WildcardType.fillList(sym.typeParams.length))
-              val subTpApprox = analyzer.approximateAbstracts(subTp) // TODO: needed?
-              // debug.patmat("subtp"+(subTpApprox <:< tpApprox, subTpApprox, tpApprox))
-              if (subTpApprox <:< tpApprox) Some(checkableType(subTp))
-              else None
-            }
-          }
+    def enumerateSubtypes(tp: Type, grouped: Boolean): List[List[Type]] = tp.typeSymbol match {
+      case    UnitClass                 => List(List(UnitTpe))
```

Comment:
kind of weird alignment

```suggestion
      case UnitClass                    => List(List(UnitTpe))
```